app/src/main/java/com/github/sdp_begreen/begreen/activities/MainActivity.kt

```diff
@@ -254,4 +262,45 @@ class MainActivity : AppCompatActivity() {
             }
         }
     }
+
+    private fun showContactUsBottomSheet() {
+
+        val bottomSheetDialog = BottomSheetDialog(this, R.style.SheetDialog)
+        bottomSheetDialog.setContentView(R.layout.bottom_sheet_contact_us)
+        val btnExit = bottomSheetDialog.findViewById<Button>(R.id.cancel_button)
+        val btnSend = bottomSheetDialog.findViewById<Button>(R.id.send_button)
+
+        val tvDate = bottomSheetDialog.findViewById<TextView>(R.id.date_textview)
+        val etMessage = bottomSheetDialog.findViewById<EditText>(R.id.message_edittext)
+
+        val date = Date() // Get the current date and time
+        val dateFormat = SimpleDateFormat("dd MMM yyyy HH:mm:ss", Locale.getDefault())
+        val formattedDate = dateFormat.format(date)
+
+        tvDate?.text = formattedDate
+
+        btnExit!!.setOnClickListener {
+            bottomSheetDialog.dismiss()
+        }
+
+        btnSend!!.setOnClickListener {
+            val msg = etMessage?.text.toString()
+            if (msg.isEmpty()) {
+                etMessage?.error = "Enter a message"
+            }
+            else {
+                firebaseDatabase.child("contact_us").child(com.google.firebase.auth.FirebaseAuth.getInstance().uid!!)
+                    .child(formattedDate).setValue(msg).addOnCompleteListener {
+                        if (it.isSuccessful) {
+                            Toast.makeText(this, "Message sent successfully", Toast.LENGTH_SHORT).show()
```

Comment:
Since this text is displayed to the user, you should add it in the string.xml ressource file. Here is the code :

In strings.xml :

`<string name="message_sent_success">Message sent successfully</string>`
`<string name="message_sent_error">Something went wrong</string>`

In this test : 

`Toast.makeText(this, R.string.message_sent_success, Toast.LENGTH_SHORT).show()`
`Toast.makeText(this, R.string.message_sent_error, Toast.LENGTH_SHORT).show()`

Reply:
Thanks for spotting that! I made the requested changes.